pkg/log/zap/zap.go

```diff
@@ -255,6 +277,14 @@ func (o *Options) BindFlags(fs *flag.FlagSet) {
 func UseFlagOptions(in *Options) Opts {
 	return func(o *Options) {
 		*o = *in
+		if o.Encoder != nil && o.TimeEncoder != nil {
+			ecfs = append(ecfs, withTimeEncoding(o.TimeEncoder))
+			if encVal.value == "json" {
+				o.Encoder = newJSONEncoder(ecfs)
+			} else {
+				o.Encoder = newConsoleEncoder(ecfs)
```

Comment:
Must `o.Encoder` always be set here? If so are we assuming the default is "console"?

Reply:
Yes, the possible values are `json` or `console`, we error out for anything else.
And the reason, we are setting here, is to set any EncoderConfigOptions that are passed using commandLine Flags, before we call `addDefaults`.